targets/psa_switch/externs/psa_random.cpp

```diff
@@ -0,0 +1,56 @@
+/* Copyright 2020-present Cornell University
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ * Yunhe Liu (yunheliu@cs.cornell.edu)
+ *
+ */
+
+#include "psa_random.h"
+#include <bm/bm_sim/_assert.h>
+
+#include <random>
+#include <thread>
+
+namespace bm {
+
+namespace psa {
+
+void
+PSA_Random::read(Data &value) {
+    int min_val = min.get_int();
+    int max_val = max.get_int();
```

Comment:
do the conversion to integers in the `init` method so it does not have to be done for every call to `read`